core/src/main/java/com/netease/arctic/log/LogDataJsonSerialization.java

```diff
@@ -42,7 +46,12 @@
 
   private transient LogDataToJsonConverters.LogDataToJsonConverter.FormatConverterContext converterContext;
 
-  public LogDataJsonSerialization(Schema schema, LogData.FieldGetterFactory<T> fieldGetterFactory) {
+  public LogDataJsonSerialization(Schema schema, FieldGetterFactory<T> fieldGetterFactory) {
+    this.schema = schema;
+    this.fieldGetterFactory = fieldGetterFactory;
+  }
+
+  public void init() {
```

Comment:
Please add annotation to remind devs of that the method should be invoked before serialize()

Reply:
Or shall we automatically initialize this variable when using it?

Reply:
> 

One way I can think of to automatically initialize this variable is to judge if the logDataToJsonConverter variable has been initialized when the serialize() method is called and initialize the variable if it has not. I wonder if you think this solution would work

Reply:
Yes, the same as I think about.